src/helpers/game_utils.ts

```diff
@@ -46,6 +48,15 @@ async function getFilteredSongList(guildPreference: GuildPreference, ignoredVide
     if (ignoredVideoIds && ignoredVideoIds.length > 0) {
         result = result.filter((song) => !ignoredVideoIds.includes(song.youtubeLink));
     }
+    if (guildPreference.getGender()[0] === GENDER.ALTERNATING && alternatingGender !== undefined) {
```

Comment:
isn't `lastAlternatingGender` initialized to null

Reply:
The undefined check here is unrelated to when `lastAlternatingGender`'s value is null. When it is null, it fails the first portion of this if statement, since we're not in alternating mode. It would be better to change the `!== undefined` to `!== null`, but when `getSongCount` (which doesn't pass a value for `alternatingGender`) is called, it will continue through this portion of the code since `null !== undefined`. Is the solution to this to change `!== undefined` to `!= null`?